File: Supplier.cs
using System;
using Newtonsoft.Json;
using System.Text.Json.Serialization;
 
namespace Project
{
    public class Supplier : SupplierShort
    {
        [JsonInclude]
        private string address;
        [JsonInclude]
        private string email;
 
        public Supplier(int?id, string name, string address, string phoneNumber, string email, string inn, string ogrn)
            : base(id, name, phoneNumber, inn, ogrn)
        {
            SetAddress(address);
            SetEmail(email);
        }
 
        public static bool ValidateAddress(string address)
        {
            return !string.IsNullOrEmpty(address) && Regex.IsMatch(address, "^[А-Яа-я\\s-]+,\\s*[А-Яа-я\\s,.\\-]+,\\s*\\d+$");
        }
 
        public static bool ValidateEmail(string email)
        {
            return !string.IsNullOrEmpty(email) && Regex.IsMatch(email, @"(.+)@(.+)\.(.+)");
        }
 
        public string GetAddress()
        {
            return address;
        }
 
        public void SetAddress(string address)
        {
            if (ValidateAddress(address))
            {
                this.address = address;
            }
            else
            {
                throw new ArgumentException("Адрес должен содержать через запятую город, улицу и дом с номером");
            }
        }
 
        public string GetEmail()
        {
            return email;
        }
 
        public void SetEmail(string email)
        {
            if (ValidateEmail(email))
            {
                this.email = email;
            }
            else
            {
                throw new ArgumentException("Неверно введён адрес электронной почты");
            }
        }
 
        public string ToFullString()
        {
            return $"ID: {id}, Name: {name}, Address: {address}, Phone: {phoneNumber}, Email: {email}, INN: {inn}, OGRN: {ogrn}";
        }
    }
}

File: Supplier.kt
class Supplier(
    private var id: Int?=null,
    private var name: String,
    private var address: String,
    private var phoneNumber: String,
    private var email: String,
    private var inn: String,
    private var ogrn: String
)
{
    fun getId(): Int? {
        return id
    }

    fun setId(id: Int?) {
        this.id = id
    }

    fun getName(): String {
        return name
    }

    fun setName(name: String) {
        this.name = name
    }

    fun getAddress(): String {
        return address
    }

    fun setAddress(address: String) {
        this.address = address
    }

    fun getPhoneNumber(): String {
        return phoneNumber
    }

    fun setPhoneNumber(phoneNumber: String) {
        this.phoneNumber = phoneNumber
    }

    fun getEmail(): String {
        return email
    }

    fun setEmail(email: String) {
        this.email = email
    }

    fun getInn(): String {
        return inn
    }

    fun setInn(inn: String) {
        this.inn = inn
    }

    fun getOgrn(): String {
        return ogrn
    }

    fun setOgrn(ogrn: String) {
        this.ogrn = ogrn
    }
}

fun main() {
    val supplier = Supplier(
        name = "Example Supplier",
        address = "456 Supplier St",
        phoneNumber = "+0987654321",
        email = "supplier@example.com",
        inn = "0987654321",
        ogrn = "0987654321098"
    )

    println(supplier.getId())
    println(supplier.getName())
    println(supplier.getAddress())
    supplier.setEmail("newemail@example.com")
    println(supplier.getEmail())
}
